### PR TITLE
fix(masthead): added tokens, new design spec

### LINKED ISSUE #5712: Consume tokens in: Masthead and Navigation
[Figma design for Masthead](https://www.figma.com/file/VMEX8Xg2nzhBX8rfBx53jp/Penta-Component-Library?type=design&node-id=3%3A33&mode=design&t=doZDRUaxE4vsn5dD-1)

[Figma design for Navigation](https://www.figma.com/file/VMEX8Xg2nzhBX8rfBx53jp/Penta-Component-Library?type=design&node-id=3%3A35&mode=design&t=doZDRUaxE4vsn5dD-1)
```[tasklist]
### Tasks
- [ ] https://github.com/patternfly/patternfly/issues/6000
- [ ] https://github.com/patternfly/patternfly/issues/5999
```

### PR DESCRIPTION
**Pull request by mattnolting**
closes #5712

Notes:
- Added `// * {Component}` to `:root` variable comments to make them searchable by {Component} or * {Component}
- Added `// - {Component}` to component/elements comments to make them searchable by {Component} or - {Component}
- Used variable fallbacks for shared values (reduces a lot of code, especially for logical prop padding updetes)
- Removed unnecessary styling
- Simplified implementation

Carrying over from https://github.com/patternfly/patternfly/pull/6059

### CODE REVIEW

**Comment by srambach** [comment]
@mattnolting Did you intend to also have the divider changes in this PR?

**Comment by mattnolting** [comment]
> @mattnolting Did you intend to also have the divider changes in this PR?

@srambach  nope, thank you. 👍

File: src/patternfly/components/Masthead/masthead-logo.hbs [hunk 1]
```diff
@@ -1,12 +1,12 @@
 <svg height="36px" viewBox="0 0 679 158" role="img">
   <title>Patternfly logo</title>
   <defs>
-    <linearGradient x1="68%" y1="2.25860997e-13%" x2="32%" y2="100%" id="linearGradient-1">
+    <linearGradient x1="68%" y1="2.25860997e-13%" x2="32%" y2="100%">
```
Comment (srambach) [inline src/patternfly/components/Masthead/masthead-logo.hbs:4]: I suspect removing this id has made the logo lose its wings - it looks like that part of the svg is looking for that id.

File: src/patternfly/components/Masthead/masthead-logo.hbs [hunk 2]
```diff
@@ -19,7 +19,7 @@
       <polygon points="399.96 65.25 399.96 2.25 414.54 0 414.54 65.25"></polygon>
       <path d="M429.21,84.69 C428.07,84.69 426.96,84.645 425.88,84.555 C424.8,84.465 423.9,84.33 423.18,84.15 L423.18,71.73 C424.38,71.97 425.88,72.09 427.68,72.09 C432.36,72.09 435.51,70.05 437.13,65.97 L437.13,65.88 L418.86,17.64 L434.97,17.64 L445.5,47.61 L457.74,17.64 L473.49,17.64 L452.16,67.68 C450.42,71.82 448.5,75.135 446.4,77.625 C444.3,80.115 441.87,81.915 439.11,83.025 C436.35,84.135 433.05,84.69 429.21,84.69 Z"></path>
   </g>
-  <g id="Logo" transform="translate(0.000000, 0.000000)">
+  <g transform="translate(0.000000, 0.000000)">
       <path d="M61.826087,0 L158,0 L158,96.173913 L147.695652,96.173913 C100.271201,96.173913 61.826087,57.7287992 61.826087,10.3043478 L61.826087,0 L61.826087,0 Z" id="Rectangle-Copy-17" fill="var(--pf-t--global--color--brand--200)"></path>
```
Comment (mattnolting) [inline src/patternfly/components/Masthead/masthead-logo.hbs:23]:
```suggestion
      <path d="M61.826087,0 L158,0 L158,96.173913 L147.695652,96.173913 C100.271201,96.173913 61.826087,57.7287992 61.826087,10.3043478 L61.826087,0 L61.826087,0 Z" fill="var(--pf-t--global--color--brand--200)"></path>
```

**Comment by srambach** [comment]
I see one difference in a divider in the mobile view but I think this can be addressed later. The overlap with the title actually exists in v5 so we can look at that later too. 
image
Looks like there's a duplicate ID causing a11y to fail?

File: src/patternfly/components/Masthead/masthead-logo.hbs [hunk 3]
```diff
@@ -1,28 +1,30 @@
...
-  </g>
-  <g id="Logo" transform="translate(0.000000, 0.000000)">
-      <path d="M61.826087,0 L158,0 L158,96.173913 L147.695652,96.173913 C100.271201,96.173913 61.826087,57.7287992 61.826087,10.3043478 L61.826087,0 L61.826087,0 Z" id="Rectangle-Copy-17" fill="var(--pf-t--global--color--brand--200)"></path>
-      <path d="M158,3.43478261 L65.2608696,158 L138,158 C149.045695,158 158,149.045695 158,138 L158,3.43478261 L158,3.43478261 Z" fill="url(#linearGradient-1)"></path>
-      <path d="M123.652174,-30.9130435 L30.9130435,123.652174 L103.652174,123.652174 C114.697869,123.652174 123.652174,114.697869 123.652174,103.652174 L123.652174,-30.9130435 L123.652174,-30.9130435 Z" fill="url(#linearGradient-1)" transform="translate(77.282609, 46.369565) scale(1, -1) rotate(90.000000) translate(-77.282609, -46.369565) "></path>
+
+<?xml version="1.0" encoding="UTF-8"?>
+<svg height="40px" viewBox="0 0 679 158" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
```
Comment (mcoker) [inline src/patternfly/components/Masthead/masthead-logo.hbs:3]: you don't need this other stuff if the svg is included on the page

```suggestion
<svg height="40px" viewBox="0 0 679 158">
```

**Review by mcoker — APPROVED** [review]

**Pull request merged**